CardScan/Classes/ScanBaseViewController.swift

```diff
@@ -184,44 +191,37 @@ public protocol TestingImageDataSource: AnyObject {
         self.videoFeed.requestCameraAccess(permissionDelegate: self)
     }
     
-    func setVideoOrientation() {
-        if ScanBaseViewController.isPadAndFormsheet {
-            self.previewView?.videoOrientation = AVCaptureVideoOrientation(rawValue: UIWindow.interfaceOrientation.rawValue)
-            self.videoFeed.videoOrientation = self.previewView?.videoOrientation
-        } else {
-            UIDevice.current.setValue(UIDeviceOrientation.portrait.rawValue, forKey: "orientation")
-            self.previewView?.videoOrientation = .portrait
-            self.videoFeed.videoOrientation = .portrait
-        }
-    }
-    
     public func setupOnViewDidLoad(regionOfInterestLabel: UIView, blurView: BlurView, previewView: PreviewView, cornerView: CornerView?, debugImageView: UIImageView?, torchLevel: Float?) {
         
         self.regionOfInterestLabel = regionOfInterestLabel
         self.blurView = blurView
         self.previewView = previewView
         self.debugImageView = debugImageView
         self.cornerView = cornerView
+        ScanBaseViewController.isPadAndFormsheet = UIDevice.current.userInterfaceIdiom == .pad && self.modalPresentationStyle == .formSheet
         
         setNeedsStatusBarAppearanceUpdate()
         regionOfInterestLabel.layer.masksToBounds = true
         regionOfInterestLabel.layer.cornerRadius = self.regionOfInterestCornerRadius
         regionOfInterestLabel.layer.borderColor = UIColor.white.cgColor
         regionOfInterestLabel.layer.borderWidth = 2.0
-  
-        self.ocrMainLoop()?.mainLoopDelegate = self
-        self.previewView?.videoPreviewLayer.session = self.videoFeed.session
-        ScanBaseViewController.isPadAndFormsheet = UIDevice.current.userInterfaceIdiom == .pad && self.modalPresentationStyle == .formSheet
-        
-        self.setVideoOrientation()
+    
+        if !ScanBaseViewController.isPadAndFormsheet {
+            UIDevice.current.setValue(UIDeviceOrientation.portrait.rawValue, forKey: "orientation")
+        }
         
         if testingImageDataSource != nil {
             self.ocrMainLoop()?.imageQueueSize = 20
         }
         
+        self.ocrMainLoop()?.mainLoopDelegate = self
+        self.previewView?.videoPreviewLayer.session = self.videoFeed.session
+        
+        self.videoFeed.videoOrientation = self.initialVideoOrientation
         self.videoFeed.pauseSession()
         //Apple example app sets up in viewDidLoad: https://developer.apple.com/documentation/avfoundation/cameras_and_media_capture/avcam_building_a_camera_app
         self.videoFeed.setup(captureDelegate: self, completion: { success in
+            self.previewView?.videoPreviewLayer.connection?.videoOrientation = self.initialVideoOrientation
```

Comment:
you should check if orientation setting is supported before setting it: https://developer.apple.com/documentation/avfoundation/avcaptureconnection/1389415-videoorientation

Reply:
Also you should probably only do this on the success case? We need to test with permissions to make sure that it's working.

Reply:
okiedokie changed. also i deleted the app and ran it again to check with the initial permissions